src/lib/routing-table.js

```diff
@@ -35,6 +35,28 @@ class RoutingTable {
     return false
   }
 
+  /**
+   * Compute a curve quote's `appliesToPrefix`, which is the shortest prefix
+   * that uniquely matches the target.
+   *
+   * @param {IlpAddress} routePrefix
+   * @param {IlpAddress} destinationAccount
+   * @returns {IlpAddress}
+   */
+  getAppliesToPrefix (routePrefix, destinationAccount) {
```

Comment:
ok, I understand now. I agree it shouldn't update the curve for "" if a separate route exists for "abc.". It should in that case store it as the curve for "random." (not "r", because that is not a ledger prefix since it doesn't end in a ".")

Maybe add a code comment explaining that in general, our routing tables assume that a route which has a curve can not have subledgers (only curveless routes can have subledgers, otherwise things get messy).